io-engine/src/eventing/mod.rs

```diff
@@ -1,8 +1,22 @@
-pub mod pool_events;
-use events_api::event::{EventAction, EventMessage};
+mod nexus_events;
+mod pool_events;
+use events_api::event::{EventAction, EventMessage, RebuildStatus};
 
 /// Event trait definition for creating events.
 pub(crate) trait Event {
     /// Create event message.
-    fn event(&self, event_action: Option<EventAction>) -> EventMessage;
+    fn event(&self, event_action: EventAction) -> EventMessage;
+}
+
+/// Rebuild event trait definition for creating rebuild events.
+pub(crate) trait RebuildEvent {
```

Comment:
A bit confused between Event and this RebuildEvent.. Seems like the rebuild_event adds more information?

I wonder if we could have something more like this:
```rust
trait Event {
    fn event(&self, event_action: EventAction) -> EventMessage;
    fn event_meta(&self, action: EventAction, meta: &impl EventMetaGen) -> EventMessage;
}
trait EventMetaGen {
   fn meta(&self)  -> EventMeta {
}

impl EventMetaGen for RebuildJob {
   fn meta(&self)  -> EventMeta {
    ....
   }
}

nexus.event_meta(Rebuild::End, &rebuild_job).generate();
```

Reply:
Made a few changes.